test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/RequestTests.cs

```diff
@@ -28,17 +40,20 @@ public class RequestTests
                     app.Run(async context =>
                     {
                         // Read the full request body
-                        var total = 0;
-                        var bytes = new byte[1024];
-                        var count = await context.Request.Body.ReadAsync(bytes, 0, bytes.Length);
-                        while (count > 0)
+                        long total = 0;
+                        var receivedBytes = new byte[bufferLength];
+                        var received = 0;
+                        while ((received = await context.Request.Body.ReadAsync(receivedBytes, 0, receivedBytes.Length)) > 0)
                         {
-                            for (int i = 0; i < count; i++)
+                            if (checkBytes)
                             {
-                                Assert.Equal(total % 256, bytes[i]);
-                                total++;
+                                for (int i = 0; i < received; i++)
```

Comment:
Could you generate the expected sequence up front and do an `Assert.Equal` instead? 

```
var contentBytes = Enumerable.Range(0, bufferLength).ToArray();
...

Assert.Equal(contentBytes, receivedBytes);
```